contracts/zero-ex/contracts/src/ZeroEx.sol

```diff
@@ -40,52 +36,66 @@ contract ZeroEx {
         // Temporarily create and register the bootstrap feature.
         // It will deregister itself after `bootstrap()` has been called.
         BootstrapFeature bootstrap = new BootstrapFeature(bootstrapper);
-        LibProxyStorage.getStorage().impls[bootstrap.bootstrap.selector] =
-            address(bootstrap);
+        mapping(bytes4 => address) storage impls =
+            LibProxyStorage.getStorage().impls;
+        impls[bootstrap.bootstrap.selector] = address(bootstrap);
+
+        // Store the slot for impls so it's accessible as a constant in Yul.
+        // Yul can't directly access immutables, so we have to go through a
+        // local.
+        uint256 implsSlot;
+        assembly { implsSlot := impls_slot }
+        immutableImplsSlot = implsSlot;
     }
 
+
     // solhint-disable state-visibility
 
     /// @dev Forwards calls to the appropriate implementation contract.
     fallback() external payable {
-        bytes4 selector = msg.data.readBytes4(0);
-        address impl = getFunctionImplementation(selector);
-        if (impl == address(0)) {
-            _revertWithData(LibProxyRichErrors.NotImplementedError(selector));
-        }
-
-        (bool success, bytes memory resultData) = impl.delegatecall(msg.data);
-        if (!success) {
-            _revertWithData(resultData);
+        // Yul can't directly access immutables.
+        uint256 implsSlot = immutableImplsSlot;
```

Comment:
Does this really give us meaningful gas savings over getting it through `LibProxyStorage.getStorage().impls`? Am I missing something? If not, I would rather we do that for simplicity.

Reply:
Good point. I'll measure it and see what the difference is.

Reply:
Result: 129 gas difference, so I got rid of the immutable in favor of clearer code.